block/manager_test.go

```diff
@@ -282,6 +296,18 @@ func getTempKVStore(t *testing.T) ds.TxnDatastore {
 	return kvStore
 }
 
+// invalidateBlockHeader results in a block header that produces a marshalling error
+func invalidateBlockHeader(block *types.Block) {
```

Comment:
As marshaling errors are not tested now, let's remove this function.

Reply:
marshaling test case should just be added back in.

Reply:
Added new tests compatible with the latest. I have also added a mock for the K-V Store using mockery.

One point to note is, before the recent updates, for the test condition: `Test_submitBlocksToDA_BlockMarshalErrorCase2`,
i.e. if A, B, and C are 3 pending blocks not yet submitted to the DA and block C has a marshal error, A and B get submitted. Now however none of them is submitted. (Rare event)

I'm using a more recent version of mockery do let me know if that's an issue.